packages/itwinui-css/src/utils/icon/icon.scss

```diff
@@ -5,17 +5,17 @@ $iui-icon-color-multicolor-disabled: grayscale(100%);
 
 $iui-icon-sizes: (
   // 12px
-  s: var(--iui-size-s),
+  's': var(--iui-size-s),
   // 16px
-  m: var(--iui-size-m),
+  'm': var(--iui-size-m),
   // 24px
-  l: calc(1.5 * var(--iui-size-m)),
+  'l': calc(1.5 * var(--iui-size-m)),
   // 32px
-  xl: var(--iui-size-xl),
+  'xl': var(--iui-size-xl),
   // 48px
-  2xl: calc(3 * var(--iui-size-m)),
+  '2xl': calc(3 * var(--iui-size-m)),
   // 64px
-  3xl: var(--iui-size-2xl)
+  '3xl': var(--iui-size-2xl)
```

Comment:
I added data attributes for this entire list of sizes that we already have defined, but i think we should narrow this down. Some of these are not used even internally by any components, and it doesn't feel right that some of these are using `calc` with magic numbers.

This was also the reason icon classes were excluded from the initial v1 release.

Reply:
@FlyersPh9

Reply:
I'd like for users to be able to pair an icon size with [each font size we offer](https://itwin.github.io/iTwinUI/typography.html).  How we do that, I am open to suggestions.

Reply:
Currently we offer these font sizes (in rem): 12px, 14px, 16px, 18px, 24px, 32px

And our current icon map has the following (in rem): 12px, 16px, 24px, 32px, 48px, 64px

In avatar, we are using: 24px, 32px, 48px, 64px
In progress-indicator, we are using: 16px, 24px, 32px, 48px
In radio-tile, we are using: 32px
In tile and file-upload, we are using: 64px
(and in a bunch of other components, we are using: 12px, 16px, 24px)

so it will not be enough to only have sizes for each font size. in fact some of it will probably not even be useful because of how close together they are (14px, 18px), and the icons also don't correspond 1:1 to the font size they need.

with that in mind, here's what i suggest:
- offer a limited set of commonly-used sizes. lets say: 12px, 16px, 24px, and maybe 32px
- consider avatar and progress-indicator as outliers. they don't need to follow the icon sizing scale.
- for any bespoke icons that don't fit the scale, suggest calculating manually.

thoughts?

Reply:
> the icons also don't correspond 1:1 to the font size they need.

I know that we like the 16px icon size + 14px font size pairing.  Any other sizes could scale accordingly to match.  I do think people will want recommended sizes (wether we provide them or not) to pair with the font sizes.

> consider avatar and progress-indicator as outliers. they don't need to follow the icon sizing scale.

Agreed.

Reply:
removed `xl`, `2xl` and `3xl`. there are now only 4 sizes: `s` (12px), `m` (16px), `l` (24px) and `auto` (1em, scales with text).

let me know what you think. looking at it 100% zoom level, i think some of these are too small
![](https://user-images.githubusercontent.com/9084735/207978088-d77e4226-54e9-43b4-a100-9d41c6b24c49.png)

Reply:
clamping it to a minimum of 1rem using this value `max(1rem, 1em)` looks like this:
![](https://user-images.githubusercontent.com/9084735/207978617-2c4842b3-a579-4cc3-be45-54d8be4147a0.png)

Reply:
`1.125rem`
![](https://user-images.githubusercontent.com/9084735/207978864-d1456d24-5666-421f-aa34-c60c9c1abefb.png)

`calc(16/14 * 1rem)`
![](https://user-images.githubusercontent.com/9084735/207979069-531e623b-89e1-4942-b80d-b56602bae309.png)

Reply:
I don't like `max(1rem, 1em)` because the small text & body text share the same icon size.

I can't see a visual difference between `1.125rem` & `calc(16/14 * 1rem)` but they both look good to me.

Reply:
now using 16/14 which equals `1.14286em`